llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1029,12 +1035,13 @@ void PassBuilder::addVectorPasses(OptimizationLevel Level,
   // convert to more optimized IR using more aggressive simplify CFG options.
   // The extra sinking transform can create larger basic blocks, so do this
   // before SLP vectorization.
-  FPM.addPass(SimplifyCFGPass(SimplifyCFGOptions()
-                                  .forwardSwitchCondToPhi(true)
-                                  .convertSwitchToLookupTable(true)
-                                  .needCanonicalLoops(false)
-                                  .hoistCommonInsts(true)
-                                  .sinkCommonInsts(true)));
+  if (!SYCLOptimizationMode)
```

Comment:
This should not be necessary. `addVectorPasses` is not called in SYCL mode.

Reply:
On main branch it is called always now for new PM. However, https://github.com/intel/llvm/pull/5305 will fix that.

Reply:
I would rather cherry-pick https://github.com/intel/llvm/pull/5305 to get relevant test status with NewPM.
Legacy pass manager does not include `addVectorPasses` in SYCL mode, so we can safely remove corresponding changes in llvm/lib/Transforms/IPO/PassManagerBuilder.cpp.

Reply:
Mikhail's patch allows to not touch this part of code here.